packages/starlight/utils/navigation.ts

```diff
@@ -224,7 +224,7 @@ function sortDirEntries(
 		// Pages are sorted by order in ascending order.
 		if (aOrder !== bOrder) return aOrder < bOrder ? -1 : 1;
 		// If two pages have the same order value they will be sorted by their slug.
-		return collator.compare(isDir(a) ? keyA : a.slug, isDir(b) ? keyB : b.slug);
+		return collator.compare(isDir(a) ? keyA : stripExtension(basename(a.entry.id)), isDir(b) ? keyB : stripExtension(basename(b.entry.id)));
```

Comment:
Couple of notes:

- We can use `a.id` and `b.id` here. These are localized for the current route, where the `entry.id` are the original ID (in the case of fallback content). Don’t think it matters too much in this case, but might as well be correct plus they’re shorter.

- I think we need to consider `index` handling here. Maybe something like:

   ```js
   function getComparisonId(id: string) {
     const basename = stripExtension(basename(id));
     return basename === 'index' ? '' : return basename;
   }
   ```
   
   Not 100% sure what the right return value is here to ensure index gets sorted first. In English, `''` works from my testing, but not impossible empty string sorting varies by locale.

Reply:
great idea! I can further test what happens in other languages.

Reply:
On second thought this fix might work wonderfully. I found a bug withtin a bug, so since the `filename` is consistent across locales. We don't really need to pass a locale based on the route. We can just use the default locale which will be, most likely, the language that author will write the filename in.